### PR TITLE
Add Windows support

### PR DESCRIPTION
**Pull request by euclio**

### CODE REVIEW

File: instant-markdown-d [hunk 1]
```diff
@@ -1,10 +1,10 @@
-#!/bin/sh
-':' //; exec "`command -v nodejs || command -v node`" "$0"
+#!/usr/bin/env node
```
Comment (suan) [inline instant-markdown-d]: Are you positive that the original lines don't work? According to http://unix.stackexchange.com/users/46546/bruno-jouhier it should...

File: instant-markdown-d [hunk 2]
```diff
@@ -59,7 +59,9 @@ function httpHandler(req, res) {
         newHTML += data;
       });
       gfm.on('exit',function(ecode){
-        socket.emit('newContent', newHTML);
+        if (socket) {
+          socket.emit('newContent', newHTML);
+        }
```
Comment (suan) [inline instant-markdown-d]: Just curious why this was necessary? Did you encounter cases where it wasn't defined?

**Comment by suan** [comment]
Pull request looks good other than my couple of comments. I'm downloading a windows VM to try out the shebang stuff

File: instant-markdown-d (hunk 1 above)
Comment (euclio) [inline instant-markdown-d]: Yeah, I know this causes a problem with the way that Ubuntu packages node, but nodejs will not generate the instant-markdown-d.cmd script necessary for running the server on Windows if the shebang is not `/usr/bin/env node`. Perhaps the solution would be to have Ubuntu users make a symlink from nodejs to node?

File: instant-markdown-d (hunk 2 above)
Comment (euclio) [inline instant-markdown-d]: Yeah, it was not easily reproducible unfortunately. However this saves the server from crashing when it is defined.

**Comment by euclio** [comment]
Updated to master. Removed the check for null socket, as I could not reproduce.

File: instant-markdown-d (hunk 1 above)
Comment (euclio) [inline instant-markdown-d]: Ah. Actually it seems the solution is to have any Ubuntu/Debian users install the nodejs-legacy package, which provides the symlink I suggested.

Comment (suan) [inline instant-markdown-d]: Yeah, I'll add that as a requirement for them going forward

**Comment by jamietre** [comment]
ooks like this died on the vine... any way this can be merged? 

The lack of windows support also makes dependencies like [https://github.com/dbankier/vscode-instant-markdown](vscode-instant-markdown) not work in windows.

**Comment by suan** [comment]
Yes! I'm setting up a vm to verify the changes and hope to have it merged soon. Nice-Had no idea that vscode plugin existed!

**Comment by jamietre** [comment]
Awesome! Thanks!

**Comment by QMonkey** [comment]
If the pull request could be merged soon, that would be better!

**Comment by suan** [comment]
Merged - thanks again!

**Comment by QMonkey** [comment]
Thanks!

**Pull request closed without merging**